lib/widgets/find_route_in_map.dart

```diff
@@ -349,130 +482,170 @@ class FindRouteInMapState extends State<FindRouteInMap>
-        double? cost;
-        try {
-          cost = firstResultRoute.getCost(attributeName: 'GreenScoreCost');
-          // cost = firstResultRoute.getCost(attributeName: `);
-        } on ArcGISException catch (_) {
-          cost = firstResultRoute.totalLength;
-        }
-        if (cost == null) {
-          _message = 'Route found, but GreenScore not available';
-        }
-        setState(() {
-          _message = 'Route found with cost: $cost + $routeCount route(s)';
-        });
-        final routeGeometry = firstResultRoute.routeGeometry;
-        _routeOverlay!.graphics.clear();
-        _routeOverlay!.graphics.add(Graphic(geometry: routeGeometry));
+      final shortestParams = await _routeTask!.createDefaultParameters();
+      shortestParams.clearStops();
+      shortestParams.setStops(stops);
+      shortestParams.travelMode = _walkingShortestMode!;
+      final shortestResult = await _routeTask!.solveRoute(shortestParams);
+      _shortestRouteOverlay?.graphics.clear();
+      double? shortestLength;
+      double shortestPercent = 0.0;
+
+      if (shortestResult.routes.isNotEmpty) {
+        final shortestRoute = shortestResult.routes.first;
+        final routeGeometry = shortestRoute.routeGeometry;
+        _shortestRouteOverlay?.graphics.add(Graphic(geometry: routeGeometry));
+        shortestLength = shortestRoute.totalLength;
+        shortestPercent = await showRoutePolygonOverlapPercent(
+          routeGeometry as Polyline,
+        );
+
+        _shortestLengthKm = shortestLength / 1000;
+        _shortestPercent = shortestPercent;
       }
-    } on ArcGISException catch (e) {
-      // If an error occurs, clear the route overlay and display the error.
-      _routeOverlay!.graphics.clear();
+
+      final greenestParams = await _routeTask!.createDefaultParameters();
+      greenestParams.clearStops();
+      greenestParams.setStops(stops);
+      greenestParams.travelMode = _walkingGreenMode!;
+      final greenestResult = await _routeTask!.solveRoute(greenestParams);
+      _greenestRouteOverlay?.graphics.clear();
+      double? greenLength;
+      double greenestPercent = 0.0;
+
+      if (greenestResult.routes.isNotEmpty) {
+        final greenestRoute = greenestResult.routes.first;
+        final routeGeometry = greenestRoute.routeGeometry;
+        _greenestRouteOverlay?.graphics.add(Graphic(geometry: routeGeometry));
+        greenLength = greenestRoute.totalLength;
+        greenestPercent = await showRoutePolygonOverlapPercent(
+          routeGeometry as Polyline,
+        );
+
+        _greenestLengthKm = greenLength / 1000;
+        _greenestPercent = greenestPercent;
+      }
+
+      setState(() {
+        if (_greenestRouteOverlay != null &&
+            _greenestRouteOverlay!.graphics.isNotEmpty) {
+          _selectedRouteType = 'greenest';
+          centerOnRoute(_greenestRouteOverlay!);
+        }
+      });
+    } on ArcGISException catch (e, stack) {
```

Comment:
```suggestion
    } on ArcGISException catch (e) {
```
Analyzer shows this as unused, so can be removed